bot/bot.go

```diff
@@ -114,6 +115,10 @@ func Channels(channels []string) func(*BotImpl) {
 	return func(b *BotImpl) { b.channels = channels }
 }
 
+func WithPrefix(prefix string) func(impl *BotImpl) {
```

Comment:
Little nit, the other options don't have `With` in front, so we could name it just `Prefix` but it's a bit unclear what kind of prefix, so I propose to name this `CommandPrefix`.

Reply:
Agreed, that reads more consistently.